parser/impl/parser.cpp

```diff
@@ -469,17 +469,163 @@ sPtr<ast::IfStatement> Parser::parseIfStatement() {
 // ---- @aabounegm
 
 sPtr<ast::Expression> Parser::parseExpression() {
-    // TODO: Concrete implementation
+    return parseBinaryExpression();
+}
+
+sPtr<ast::Expression> Parser::parseUnaryExpression() {
+    Token tok = m_lexer.Peek();
+    ast::UnaryExpression expr;
+    if (opPrec(tok.type) >= 0) {
+        // if has operations on primary
+        if (tok.type == TokenType::Not || tok.type == TokenType::Sub ||
+            tok.type == TokenType::Add) {
+            // math unary operations
+            tok = m_lexer.Next();
+            expr.operation = tok;
+            expr.operand = parseUnaryExpression();
+            return std::make_shared<ast::UnaryExpression>(expr);
+
+        } else if (tok.type == TokenType::OpenParen) {
+            // parenthesis -> more priority
+            tok = m_lexer.Next();
+            sPtr<ast::Expression> expr = parseBinaryExpression(0);
```

Comment:
```suggestion
            sPtr<ast::Expression> expr = parseBinaryExpression();
```
This will confuse readers less.